utils/python/CIME/hist_utils.py

```diff
@@ -0,0 +1,175 @@
+            shutil.copy(test_hist, new_file)
+
+    expect(num_moved > 0, "move failed: no hist files found in rundir '%s'" % rundir)
+
+    return comments
+
+def _compare_hists(case, suffix1="", suffix2="", from_dir1=os.getcwd(), from_dir2=os.getcwd()):
+
+    testcase = case.get_value("CASE")
+    all_success = True
+    num_compared = 0
+    comments = "Comparing hists for case '%s' dir1='%s', suffix1='%s',  dir2='%s' suffix2='%s'\n" % \
+        (testcase, from_dir1, suffix1, from_dir2, suffix2)
+
+    for model in _iter_model_file_substrs(case):
+        comments += "  comparing model '%s'\n" % model
+        hists1 = _get_all_hist_files(testcase, model, suffix1, from_dir1)
+        hists2 = _get_all_hist_files(testcase, model, suffix2, from_dir2)
+        len_hist1 = len(hists1)
+        len_hist2 = len(hists2)
+        if len_hist1 == 0:
+            comments += " no hist files found for model %s\n"%model
+            continue
+        if len_hist1 != len_hist2:
+            comments += "    num hists does not match %d != %d\n" % (len_hist1, len_hist2)
+            all_success = False
+            continue
+
+        num_compared += len(hists1)
+        for hist1, hist2 in zip(hists1, hists2):
+            success, cprnc_comments = cprnc(hist1, hist2, case, from_dir1)
+            if success:
+                comments += "    %s matched %s\n" % (hist1, hist2)
+            else:
+                comments += "    %s did NOT match %s\n" % (hist1, hist2)
+                comments += cprnc_comments + "\n"
+                all_success = False
+
+    expect(num_compared > 0, "Did not compare any hist files for suffix1='%s' suffix2='%s', dir1='%s', dir2='%s'\nComments=%s" %
+           (suffix1, suffix2, from_dir1, from_dir2, comments))
+    return all_success, comments
+
+def compare_test(case, suffix1, suffix2):
+    """
+    Compares two component history files in the testcase directory
+
+    returns (SUCCESS, comments)
+    """
+    rundir   = case.get_value("RUNDIR")
+
+    return _compare_hists(case, suffix1, suffix2, rundir, rundir)
+
+def cprnc(file1, file2, case, rundir):
+    cprnc_exe = case.get_value("CCSM_CPRNC")
+    basename = os.path.basename(file1)
+    stat, out, _ = run_cmd("%s -m %s %s 2>&1 | tee %s/%s.cprnc.out" % (cprnc_exe, file1, file2, rundir, basename))
+    return (stat == 0 and "files seem to be IDENTICAL" in out, out)
+
+def compare_baseline(case, baseline_dir=None):
+    """
+    compare the current test output to a baseline result
+    """
+    rundir   = case.get_value("RUNDIR")
+    if baseline_dir is None:
+        baselineroot = case.get_value("BASELINE_ROOT")
+        basecmp_dir = os.path.join(baselineroot, case.get_value("BASECMP_CASE"))
+        dirs_to_check = (baselineroot, basecmp_dir)
+    else:
+        basecmp_dir = baseline_dir
+        dirs_to_check = (basecmp_dir,)
+
+    for bdir in dirs_to_check:
+        if not os.path.isdir(bdir):
+            return False, "ERROR %s does not exist" % bdir
```

Comment:
Can this message (which I assume is added to the TestStatus file) be changed to something more uniquely greppable, like: "ERROR %s baseline directory does not exist"? Then I can do a grep -v 'baseline directory does not exist', which I imagine is less likely to occur elsewhere than simply 'does not exist'.